src/repl/scala/tools/nsc/interpreter/ReplReporter.scala

```diff
@@ -105,9 +146,22 @@ class  ReplReporter(intp: IMain) extends ConsoleReporter(intp.settings, Console.
   }
 
   override def displayPrompt(): Unit =
-    if (!intp.totalSilence) super.displayPrompt()
+    if (!totalSilence) super.displayPrompt()
 
   override def rerunWithDetails(setting: reflect.internal.settings.MutableSettings#Setting, name: String) =
     s"; for details, enable `:setting $name' or `:replay $name'"
 
 }
+
+class ReplStrippingWriter(config: ReplReporterConfig) extends PrintWriter(config.out) {
+  override def write(str: String) = super.write(truncate(strip(str)))
+
+  import config._
+  private def strip(str: String): String =
+    if (unwrapStrings) unmangle(str) else str
+
+  private def truncate(str: String): String =
+    if (isTruncating && (maxPrintString != 0 && str.length > maxPrintString))
+      (str take maxPrintString - 3) + "..."
+    else str
+}
```

Comment:
A StrippingWriter is when I fantasize about J. K. Rowling. Who is also a PrintWriter, a writer in print, pant pant.

I'd thumbsup `DemanglingWriter` with similar verbs like `demangle`, maybe by the third step in modularizing REPL. I mean WREPL. Where the W is for wrapped writer, of course.